xbmc/network/NetworkServices.cpp

```diff
@@ -829,12 +829,9 @@ bool CNetworkServices::StartUPnP()
   ret |= StartUPnPClient();
   if (CSettings::GetInstance().GetBool(CSettings::SETTING_SERVICES_UPNPSERVER))
   {
-   ret |= StartUPnPServer();
-  }
-
-  if (CSettings::GetInstance().GetBool(CSettings::SETTING_SERVICES_UPNPCONTROLLER))
-  {
-    ret |= StartUPnPController();
+    ret |= StartUPnPServer(); 
+    if (CSettings::GetInstance().GetBool(CSettings::SETTING_SERVICES_UPNPCONTROLLER))
+      ret |= StartUPnPController();
   }
 
   if (CSettings::GetInstance().GetBool(CSettings::SETTING_SERVICES_UPNPRENDERER))
```

Comment:
Those two ifs are nested as to have the "controller needs server" behavior, and I think those curly braces are needed as the if block isn't one instruction long.